app/src/main/java/org/jellyfin/androidtv/ui/search/TextSearchFragment.kt

```diff
@@ -3,78 +3,92 @@ package org.jellyfin.androidtv.ui.search
 import android.os.Bundle
 import android.text.Editable
 import android.text.TextWatcher
-import android.view.KeyEvent
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.view.inputmethod.EditorInfo
 import android.widget.EditText
-import android.widget.TextView
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
 import androidx.leanback.app.RowsSupportFragment
-import androidx.leanback.widget.BaseOnItemViewClickedListener
-import androidx.leanback.widget.ListRow
-import androidx.leanback.widget.OnItemViewSelectedListener
-import org.jellyfin.androidtv.R
-import org.jellyfin.androidtv.data.service.BackgroundService
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import org.jellyfin.androidtv.databinding.FragmentSearchTextBinding
-import org.jellyfin.androidtv.ui.itemhandling.BaseRowItem
-import org.jellyfin.androidtv.ui.itemhandling.ItemLauncher
-import org.jellyfin.androidtv.ui.itemhandling.ItemRowAdapter
-import org.koin.java.KoinJavaComponent.inject
 
-class TextSearchFragment : Fragment(), TextWatcher, TextView.OnEditorActionListener {
-	private lateinit var searchProvider: SearchProvider
-	private val backgroundService = inject<BackgroundService>(BackgroundService::class.java)
+class TextSearchFragment : Fragment() {
 
-	override fun onCreate(savedInstanceState: Bundle?) {
-		super.onCreate(savedInstanceState)
+	private val viewModel by viewModels<SearchViewModel>()
 
-		// Create provider
-		searchProvider = SearchProvider(requireContext(), lifecycle)
-	}
+	private var _binding: FragmentSearchTextBinding? = null
+	private val binding get() = _binding!!
```

Comment:
```suggestion
	private lateinit var binding: FragmentSearchTextBinding
```
We normally do it like this

Reply:
According to google

```
Note: Fragments outlive their views. Make sure you clean up any references to the binding class instance in the fragment's [onDestroyView()](https://developer.android.com/reference/kotlin/androidx/fragment/app/Fragment#ondestroyview) method.
```

https://developer.android.com/topic/libraries/view-binding#fragments

But if you still want to change it I will

Reply:
No thanks for linking, we should probably change our fragments to mirror this approach. I believe this wasn't in the Android docs when we started using view binding.

Reply:
FWIW I've never seen a memory leak happen because the `viewBinding` wasn't cleared, but I guess it's possible somehow